src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -220,6 +225,20 @@ private void validateTable(TableModel table, Errors errors, SchemaValidationCont
         }
     }
 
+    private void validateDataTypes(List<ColumnModel> columns, Errors errors) {
+        List validDataTypes = new ArrayList(Arrays.asList(TableDatatypes.values()));
+        for (ColumnModel column : columns) {
+            // spring defaults user input not belonging to the TableDataTypes enum to null
+            if (column.getDatatype() == null || !validDataTypes.contains(column.getDatatype())) {
+                errors.rejectValue("schema", "InvalidDatatype",
+                    "invalid datatype(s) in table columns, valid DataTypes are " +
+                        validDataTypes.stream().map(datatype -> datatype.toString()).collect(toList()));
```

Comment:
If you have an array type, you can get the string version of it by using `Arrays.toString()`
```suggestion
                        Arrays.toString(validDataTypes));
```